File: user-service/src/main/java/com/chong/pay/userservice/domain/Exchange.java
package com.chong.pay.userservice.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
@Document
public class Exchange {
    public enum EXCHANGE_TYPE{
        PAYMENT, SEND
    }

    @Id
    private String paymentId;
    private EXCHANGE_TYPE exchangeType;
    private String myId;
    private String otherId;
    private long money;     // 음수이면 송금, 양수이면 받음

//    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime exchangeDate;
    private boolean isComplete;
}

File: transfer-service/src/main/kotlin/com/chong/pay/transferservice/domain/Exchange.kt
package com.chong.pay.transferservice.domain

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

enum class EXCHANGE_TYPE {
    PAYMENT, SEND
}

@Document
data class Exchange(
        @Id
        var paymentId: String = "",
        var exchangeType: EXCHANGE_TYPE? = null,
        var myId: String = "",
        var otherId: String = "",
        var money: Long = 0,
        @JsonSerialize(using = LocalDateTimeSerializer::class)
        @JsonDeserialize(using = LocalDateTimeDeserializer::class)
//        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
        var exchangeDate: LocalDateTime? = null,
        var isComplete: Boolean = false
)